misk-gcp-testing/src/main/kotlin/misk/cloud/gcp/spanner/GoogleSpannerEmulator.kt

```diff
@@ -124,11 +125,13 @@ class GoogleSpannerEmulator @Inject constructor(
   /**
    * Pulls a Docker container containing the Google Spanner emulator.
    */
-  fun pullImage() {
+  fun pullImage(imageVersion: String? = null) {
+    setImageVersion(imageVersion)
     Companion.pullImage()
```

Comment:
What do you think of returning the image instead of setting it?

```suggestion
    Companion.pullImage(fullImageName(imageVersion))
```

Same with `doStart(...)`:

```
  private fun doStart(imageVersion: String? = null) {
    image = fulllImageName(imageVersion)
    ...
```

Then, below, update the function to return:

```
  private fun fullImageName(imageVersion: String?) {
    return "$IMAGE_NAME:${imageVersion ?: "latest"}"
  }
```

Reply:
I think that makes sense in the `doStart` but I think it would be confusing if in the following case
```
pullImage("1.0.0") // pulls 1.0.0
doStart() // starts latest
```

I think we would want
```
pullImage("1.0.0") // pulls 1.0.0
doStart() // starts 1.0.0
```

Ill update the `doStart` to not set the image but not the `pullImage`.  In  `doStart` I'll default to using the provided version if there is one and the set version if there is not